App/controllers/upload-file.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+var bluebird = require('bluebird');
+var xlsx = require('xlsx');
+var mongoose = require('mongoose');
+
+var awsUpload = require('../services/aws-services.js');
+var dbServices = require('../services/db-services.js');
+var jsonDataModel = require('../models/json-file-schema.js');
+
+var k = require('../../config/constants.js');
+mongoose.connect(k.DB.URL);
+var db = mongoose.connection;
+db.on('error', console.error.bind(console, 'Connection error : '));
+
+var processFileUpload = function (uploadFile) {
+  var workbook = xlsx.readFile(uploadFile.path);
+  bluebird.map(workbook.SheetNames, function (item) {
+    var jsonArray = xlsx.utils.sheet_to_json(workbook.Sheets[item]);
+    return bluebird.map(jsonArray, function (obj) {
+
+      if (typeof obj !== 'undefined' && typeof obj[k.CATEGORY_ID] !== 'undefined' && typeof obj.Category !==
+        'undefined' && typeof obj[k.META_TITLE] !== 'undefined' && typeof obj[k.META_DESCRIPTION] !==
+        'undefined') {
```

Comment:
For checking whether a variable is undefined, the method we use is `if (a)`. This checks if variable `a` is false/undefined. Use the same method. This is specified in the standards document.